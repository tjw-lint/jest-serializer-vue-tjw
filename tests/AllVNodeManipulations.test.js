const helpers = require('./helpers.js');

import { shallowMount } from '@vue/test-utils';
import AllVNodeManipulations from './components/AllVNodeManipulations.vue';

describe('AllVNodeManipulations.vue', () => {
  test('Attribute and input value shown', () => {
    helpers.mockSettings({
      addInputValues: true,
      stringifyObjects: true
    });

    const wrapper = shallowMount(AllVNodeManipulations);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Attribute value shown, input value hidden', () => {
    helpers.mockSettings({
      addInputValues: false,
      stringifyObjects: true
    });

    const wrapper = shallowMount(AllVNodeManipulations);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Attribute value hidden, input value shown', () => {
    helpers.mockSettings({
      addInputValues: true,
      stringifyObjects: false
    });

    const wrapper = shallowMount(AllVNodeManipulations);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Attribute and input value not shown', () => {
    helpers.mockSettings({
      addInputValues: false,
      stringifyObjects: false
    });

    const wrapper = shallowMount(AllVNodeManipulations);

    expect(wrapper)
      .toMatchSnapshot();
  });
});
