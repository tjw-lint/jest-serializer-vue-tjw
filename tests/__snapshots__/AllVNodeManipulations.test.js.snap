// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllVNodeManipulations.vue Attribute and input value not shown 1`] = `
<div>
  <h1 title="[object Object]">
    H1
  </h1>
  <input>
</div>
`;

exports[`AllVNodeManipulations.vue Attribute and input value shown 1`] = `
<div>
  <h1 title="{a:{b:34},c:[[],3,true]}">
    H1
  </h1>
  <input value="'moo'">
</div>
`;

exports[`AllVNodeManipulations.vue Attribute value hidden, input value shown 1`] = `
<div>
  <h1 title="[object Object]">
    H1
  </h1>
  <input value="'moo'">
</div>
`;

exports[`AllVNodeManipulations.vue Attribute value shown, input value hidden 1`] = `
<div>
  <h1 title="{a:{b:34},c:[[],3,true]}">
    H1
  </h1>
  <input>
</div>
`;
